### PR TITLE
[CI] Add version database update in CD

### PR DESCRIPTION
**Pull request by Leonz5288**
In release.yml, we will automatically update version database in every production release. We will need to add two new secrets to the repository though.

### CODE REVIEW

File: misc/save_new_version.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    'major': major,
+    'minor': minor,
+    'patch': patch,
+    'date': release_date
+}
+
+username = os.getenv('METADATA_USERNAME')
+password = os.getenv('METADATA_PASSWORD')
+
+response = requests.post('http://54.90.48.192/add_version/main',
```
Comment (rexwangcc) [inline misc/save_new_version.py:25]: How about making the IP address (i.e. `http://54.90.48.192`) an ENV VAR (e.g. `METADATA_SERVICE_URL`) as well so later if we update it to something like HTTPS://DOMAIN we don't need to change the code again? Same for the other place.

Comment (rexwangcc) [reply misc/save_new_version.py:25]: Side note: could you add all newly created credentials (username, password, url, etc.) to Vault? (and we may let Action authenticate with Vault in the future) Thanks!

File: misc/save_new_version.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                         json=payload,
+                         auth=requests.auth.HTTPBasicAuth(username, password))
+print(response.text)
```
Comment (rexwangcc) [inline misc/save_new_version.py:28]: Curious: what would happen if the response code is not 200/201? Do we need a `response.raise_for_status()` and/or try catch here? Same question applies to the other place.

**Review by rexwangcc — COMMENTED** [review]
Thanks! LGTM in general, I added a few comments.

File: misc/save_new_version.py (hunk 1 above)
Comment (Leonz5288) [reply misc/save_new_version.py:25]: Added to vault

Comment (Leonz5288) [reply misc/save_new_version.py:25]: > How about making the IP address (i.e. `http://54.90.48.192`) an ENV VAR (e.g. `METADATA_SERVICE_URL`) as well so later if we update it to something like [HTTPS://DOMAIN](HTTPS://DOMAIN) we don't need to change the code again? Same for the other place.

Good point.

File: python/build.py [hunk 3]
```diff
@@ -81,6 +81,15 @@ def parse_args():
...
     return parser.parse_args()
 
+def upload_taichi_version():
+    username = os.getenv('METADATA_USERNAME')
+    password = os.getenv('METADATA_PASSWORD')
+    filename = os.listdir('../dist')[0]
+    filename = filename[:len(filename)-4]
+    parts = filename.split('-')
+    payload = {'version': parts[1], 'platform': parts[4], 'python': parts[2]}
+    response = requests.post('http://54.90.48.192/add_version/detail', json=payload, auth=requests.auth.HTTPBasicAuth(username, password))
```
Comment (yihong0618) [inline python/build.py]: for HTTPBasicAuth 
auth=(username, password)  is fine

Comment (Leonz5288) [reply python/build.py]: Thanks!

File: python/build.py [hunk 4]
```diff
@@ -82,6 +84,38 @@ def parse_args():
...
+def upload_taichi_version():
+    username = os.getenv('METADATA_USERNAME')
+    password = os.getenv('METADATA_PASSWORD')
+    url = os.getenv('METADATA_URL')
+    filename = os.listdir('../dist')[0]
+    filename = filename[:len(filename) - 4]
+    parts = filename.split('-')
+    payload = {'version': parts[1], 'platform': parts[4], 'python': parts[2]}
+    try:
+        response = requests.post('http://' + url + '/add_version/detail',
```
Comment (rexwangcc) [inline python/build.py:96]:
```suggestion
        response = requests.post(f'http://{url}/add_version/detail',
```

File: misc/save_new_version.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    'patch': patch,
+    'date': release_date
+}
+
+username = os.getenv('METADATA_USERNAME')
+password = os.getenv('METADATA_PASSWORD')
+url = os.getenv('METADATA_URL')
+
+try:
+    response = requests.post('http://' + url + '/add_version/main',
```
Comment (rexwangcc) [inline misc/save_new_version.py:27]:
```suggestion
    response = requests.post(f'http://{url}/add_version/main',
```

**Review by rexwangcc — APPROVED** [review]
LGTM!

**Pull request merged**